pkg/api/client.go

```diff
@@ -149,7 +167,57 @@ type roundTripper struct {
 // RoundTrip implements `http.RoundTripper`
 func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	req.Header.Set("User-Agent", rt.UserAgent)
-	return rt.RoundTripper.RoundTrip(req)
+
+	// If no logging is needed, return early.
+	if !Enabled(DebugLog) {
+		return rt.RoundTripper.RoundTrip(req)
+	}
+
+	// From https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/transport/logger.go
+
+	DebugLog.Printf("--> %s %s", req.Method, req.URL)
+	// Save these headers so we can redact Authorization.
+	savedHeaders := req.Header.Clone()
+	if req.Header != nil && req.Header.Get("authorization") != "" {
+		req.Header.Set("authorization", "<redacted>")
+	}
+
+	b, err := httputil.DumpRequestOut(req, true)
+	if err == nil {
+		DebugLog.Println(string(b))
```

Comment:
Looks like codeql thinks this contains sensitive info even though you redact right above - is there a way to suppress the codeql warning?

Reply:
I've redacted the body of the request too now. This makes the logging less useful, so maybe we can find some middle ground, but I want to see if that's the issue.

Would you be able to kick off the CI, I'm not able to since I haven't committed to this repo before.

Reply:
Kicked it, still seems upset about the URL...

Reply:
This error comes from c# - https://codeql.github.com/codeql-query-help/csharp/cs-log-forging/

It looks like we're running the go version of codeql though, so not sure why we're seeing this